addons/mass_mailing_sms/models/mailing_mailing.py

```diff
@@ -51,14 +54,22 @@ def _compute_body_plaintext(self):
                 mailing.body_plaintext = mailing.sms_template_id.body
 
     @api.depends('mailing_trace_ids.failure_type')
-    def _compute_sms_has_insufficient_credit(self):
-        mailing_ids = self.env['mailing.trace'].sudo().search([
-            ('mass_mailing_id', 'in', self.ids),
-            ('trace_type', '=', 'sms'),
-            ('failure_type', '=', 'sms_credit')
-        ]).mapped('mass_mailing_id')
-        for mailing in self:
-            mailing.sms_has_insufficient_credit = mailing in mailing_ids
+    def _compute_sms_has_iap_failure(self):
+        failures = ['sms_acc', 'sms_credit'] 
+        traces = self.env['mailing.trace'].sudo().read_group([
```

Comment:
We could skip the read_group if not self.ids (saving one query when creating the mass mailing :D FP likes that ) .

Reply:
good to know :) I'd just go with something like
`if not self.ids:`
`             self.sms_has_insufficient_credit = self.sms_has_unregistered_account = False`
Would it be okay for you?

Reply:
Ok !